prow/cmd/deck/main.go

```diff
@@ -116,6 +121,39 @@ func handleData(ja *JobAgent) http.HandlerFunc {
 
 type logClient interface {
 	GetJobLog(job, id string) ([]byte, error)
+	// Add ability to stream logs with options enabled. This call is used to follow logs
+	// using kubernetes client API. All other options on the Kubernetes log api can
+	// also be enabled.
+	GetJobLogStream(job, id string, options map[string]string) (io.ReadCloser, error)
+}
+
+func httpChunking(log io.ReadCloser, w http.ResponseWriter) {
+	flusher, ok := w.(http.Flusher)
```

Comment:
flusher will be a nil interface if this coercion fails, meaning that flusher.Flush() will panic

Reply:
thanks, I am now checking for nil.